.github/workflows/python.yml

```diff
@@ -0,0 +1,57 @@
+name: Python
+
+on:
+  push:
+    branches: ["master", "ci-test"]
+    paths: ["**/*.py", "Pipfile.lock", "pyproject.toml"]
+  pull_request:
+    branches: ["master", "ci-test"]
+    paths: ["**/*.py", "Pipfile.lock", "pyproject.toml"]
+
+jobs:
+  pyright:
+    name: Pyright
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "3.11"
+          cache: pipenv
+
+      - name: Install pipenv
+        run: pipx install pipenv
```

Comment:
Could've used just `pip` since that still works, but due to PEP 668 it'll stop working at some point.